lib/puppet/parser/scope.rb

```diff
@@ -738,17 +821,27 @@ def to_s
 
   alias_method :inspect, :to_s
 
-  # remove ephemeral scope up to level
-  # TODO: Who uses :all ? Remove ??
+  # Pop ephemeral scopes up to level and return them
   #
+  # @param level [Fixnum] a positive integer
+  # @return [Array] the removed ephemeral scopes
+  # @api private
   def unset_ephemeral_var(level=:all)
     if level == :all
+      Puppet.deprecation_warning('Calling Parser::Scope#unset_ephemeral_var() with parameter :all is deprecated')
       @ephemeral = [ MatchScope.new(@symtable, nil)]
     else
       @ephemeral.pop(@ephemeral.size - level)
     end
   end
 
+  # Push ephemeral scopes onto the ephemeral scope stack
+  # @param ephemeral_scopes [Array]
+  # @api private
+  def restore_ephemerals(ephemeral_scopes)
```

Comment:
The name of the method does not rhyme with the documentation and implementation - this does not look like some kind of restore operation, it just adds ephemerals.
How about calling this `push_ephemerals`, and create an alias `pop_ephemerals` for the weird `unset_ephemeral_var` (we keep that method intact for backwards compatibility reasons).